public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -550,6 +549,27 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
         });
     }
 
+    if (config.featureToggles.influxdbBackendMigration && this.access === 'proxy') {
+      const target: InfluxQuery = {
+        refId: 'metricFindQuery',
+        query: 'SHOW TAG KEYS',
+        rawQuery: true,
+      };
+      return lastValueFrom(super.query({ targets: [target] } as DataQueryRequest))
+        .then((res: DataQueryResponse) => {
+          if (!res || !res.data || res.state !== LoadingState.Done) {
+            return { status: 'error', message: 'Error reading InfluxDB' };
+          }
+          if (res.data?.length) {
+            return { status: 'success', message: 'Data source is working' };
+          }
+          return { status: 'error', message: 'Data source is not working' };
```

Comment:
nit. In most cases, Messages like `'Data source is not working'` / `'Error reading InfluxDB'` won't helpful much.